content/3_Silver/Intro_Sorted_Sets.mdx

```diff
@@ -125,13 +123,39 @@ which can only be used once before calling the `next()` method.
 </Warning>
 
 </JavaSection>
+<PySection>
+
+In Python, we can use `iter()` to obtain the iterator object of any iterable.
+This returns the first element, and using `next()` on the iterator gives you the
+next one. Below, a dictionary is used instead of a set because dictionaries keep
+order.
+
+```py
+nums = {1: None, 7: None, 0: None, 2: None, 4: None}
+iterator = iter(nums)
+
+print(iterator)  # <set_iterator object at 0x14765f3eb6c0>
```

Comment:
is this needed lol

Reply:
I mean, for loops pretty much do the same thing in most cases. But in some cases, eg. getting any/first element of a set is easier.

```py
a = {4, 2, 3, 1}

print(next(iter(a))) # 1

for i in a:
    print(i) # 1
    break
```

From my research, it's also useful for OOP, but I haven't really experienced that myself.
Python has more useful iterators batched in the `itertools` library (which itself uses `iter()` a lot in it's functions).

Anyhow, for loops actually call on iter() and next(), so if you need more control in your iterations, it's useful.
Usually just for loops and while loops suffice though. If needed I can go over more useful iteration tools in the `itertools` library.

Reply:
yeah sure, but in cp this knowledge doesn't come into play

Reply:
Itertools is very useful for cp, whereas iter() is just the fundamentals of Python iterators. I'll prob put more focus on itertools then.